### PR TITLE
project setup

### PR DESCRIPTION
**Pull request by Boorhan**
Project setup done.

### CODE REVIEW

File: HELP.md
Comment (rummansynergy) [inline HELP.md:1]: Rename this file to README.md

File: HELP.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+* [Spring Data JPA](https://docs.spring.io/spring-boot/docs/3.3.3/reference/htmlsingle/index.html#data.sql.jpa-and-spring-data)
+* [Spring Web](https://docs.spring.io/spring-boot/docs/3.3.3/reference/htmlsingle/index.html#web)
+* [Spring Boot DevTools](https://docs.spring.io/spring-boot/docs/3.3.3/reference/htmlsingle/index.html#using.devtools)
+* [Thymeleaf](https://docs.spring.io/spring-boot/docs/3.3.3/reference/htmlsingle/index.html#web.servlet.spring-mvc.template-engines)
+* [Spring Security](https://docs.spring.io/spring-boot/docs/3.3.3/reference/htmlsingle/index.html#web.security)
+
+### Guides
+The following guides illustrate how to use some features concretely:
+
+* [Accessing Data with JPA](https://spring.io/guides/gs/accessing-data-jpa/)
```
Comment (rummansynergy) [inline HELP.md]: Remove the auto generated lines. Add relevant readme for this project.

File: mvnw
Comment (rummansynergy) [inline mvnw:1]: What's the purpose of this file?

File: mvnw.cmd [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+} catch {
+  if (! (Test-Path -Path "$MAVEN_HOME" -PathType Container)) {
+    Write-Error "fail to move MAVEN_HOME"
+  }
+} finally {
+  try { Remove-Item $TMP_DOWNLOAD_DIR -Recurse -Force | Out-Null }
+  catch { Write-Warning "Cannot remove $TMP_DOWNLOAD_DIR" }
+}
+
+Write-Output "MVN_CMD=$MAVEN_HOME/bin/$MVN_CMD"
```
Comment (rummansynergy) [inline mvnw.cmd:149]: ditto.

N.B: Ditto is used to indicate that something already said is applicable a second time.

File: src/main/resources/application.properties
Comment (rummansynergy) [inline src/main/resources/application.properties:1]: Can we convert this to yml?

**Review by rummansynergy — CHANGES_REQUESTED** [review]
Couldn't find any Dockerfile and gitignore file in the codebase.

File: target/classes/com/demo/classroom/ClassroomApplication.class
Comment (rummansynergy) [inline target/classes/com/demo/classroom/ClassroomApplication.class:1]: What's the purpose of this file?

File: target/test-classes/com/demo/classroom/ClassroomApplicationTests.class
Comment (rummansynergy) [inline target/test-classes/com/demo/classroom/ClassroomApplicationTests.class:1]: ditto

**Review by rummansynergy — CHANGES_REQUESTED** [review]

File: HELP.md
Comment (Boorhan) [reply HELP.md:1]: Done.

File: HELP.md (hunk 1 above)
Comment (Boorhan) [reply HELP.md]: Done.

File: mvnw
Comment (Boorhan) [reply mvnw:1]: The Maven Wrapper script for Unix-based systems. 

N.B: Maven Wrapper ensures that the correct version of Maven is used to build the project, regardless of what is installed on the developer’s machine or in the build environment.

File: mvnw.cmd (hunk 2 above)
Comment (Boorhan) [reply mvnw.cmd:149]: The Maven Wrapper script for Windows systems.

File: src/main/resources/application.properties
Comment (Boorhan) [reply src/main/resources/application.properties:1]: Yes, we can. Spring Boot supports both properties and YAML formats for configuration.

File: target/classes/com/demo/classroom/ClassroomApplication.class
Comment (Boorhan) [reply target/classes/com/demo/classroom/ClassroomApplication.class:1]: Also known as application entry point, serves for crucial purposes like setting up and running a Spring Boot application.

File: target/test-classes/com/demo/classroom/ClassroomApplicationTests.class
Comment (Boorhan) [reply target/test-classes/com/demo/classroom/ClassroomApplicationTests.class:1]: Mainly used to test whether the Spring application context loads successfully. This file typically serves as a foundation for writing tests for the application.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+### VS Code ###
+.vscode/
```
Comment (rummansynergy) [inline .gitignore]: Add a newline at the end.

File: README.md [hunk 4]
```diff
@@ -1 +1,52 @@
...
+
+- **`templates` Folder**: Used for server-side rendering with template engines like Thymeleaf, FreeMarker, etc.
+
+- **`pom.xml` (Maven)**: The build configuration file for Maven, defining dependencies, plugins, and other build instructions.
+
+- **Target or Build Directory**: Generated when you build the project, containing compiled classes and resources, and any JAR or WAR files generated by the build process.
+
+### Other Common Files
+- **`README.md`**: A markdown file where you document your project.
+- **`.gitignore`**: Specifies files or directories that should not be tracked by Git (e.g., `target/`, `.idea/`, etc.).
```
Comment (rummansynergy) [inline README.md]: Please add description, features, dependency and installation guide.

**Review by rummansynergy — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties
Comment (Boorhan) [reply src/main/resources/application.properties:1]: Changed the file extension and the corresponding format within it

File: .gitignore (hunk 3 above)
Comment (Boorhan) [reply .gitignore]: Done!

File: README.md (hunk 4 above)
Comment (Boorhan) [reply README.md]: Readme Updated.

File: README.md [hunk 5]
```diff
@@ -1 +1,76 @@
-# Classroom-Spring-Boot-Project
\ No newline at end of file
+# Classroom Spring Boot Prtoject
+
```
Comment (rummansynergy) [inline README.md]:
```suggestion
# Classroom Spring Boot Project

```

File: README.md [hunk 6]
```diff
@@ -1 +1,76 @@
...
+  - Suggest relevant books and materials for each course.
+  
+- **Student Features:**
+  - Enroll in multiple courses offered by different teachers.
+  - Access suggested books and resources.
+
+- **General Features:**
+  - Different login and registration processes for teachers and students.
+  - Separate dashboards for both teachers and students after login.
+
```
Comment (rummansynergy) [inline README.md]: Move the general feature to the top

**Review by rummansynergy — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (Boorhan) [reply README.md]: Done.

File: README.md (hunk 5 above)
Comment (Boorhan) [reply README.md]: Done.

**Review by rummansynergy — APPROVED** [review]
LGTM

**Pull request merged**